stdlib/LinearAlgebra/src/qr.jl

```diff
@@ -906,3 +906,14 @@ end
 ## Lower priority: Add LQ, QL and RQ factorizations
 
 # FIXME! Should add balancing option through xgebal
+
+# Compute `det` from the number of Householder reflections.
+det(Q::QRPackedQ) = _det_qr(Q.factors, Q.τ)
+_det_qr(factors, τ) =
+    prod(axes(τ, 1)) do i
+        Base.@_inline_meta
+        # "v" in the math definition except for the diagonal (= 1):
+        v = @inbounds @view factors[i + 1:end, i]
+        𝟙 = one(eltype(factors))
+        𝟙 - (𝟙 + v'v) * @inbounds τ[i]
```

Comment:
FYI, it's much faster (~ 1.5x) to directly implement it using `for` loop as below for moderate-size matrices (~ 100).  The difference is bigger (~ 2.4x) without `@inline` annotation added in 940d16c85d5231a6bd3235f789fcef8968bf266f.

```julia
_det_qr(factors, τ) =
    prod(axes(τ, 1)) do i
        Base.@_inline_meta
        vv = 𝟙 = one(eltype(factors))
        @simd for j in axes(factors, 1)[i + 1:end]
            vv += abs2(@inbounds factors[j, i])
        end
        𝟙 - vv * @inbounds τ[i]
    end
```

But I thought it'd be much better to tweak `v'v` in such a way that the code like above is unnecessary.

Reply:
I agree that the current version looks better